src/llmcompressor/modifiers/quantization/calibration.py

```diff
@@ -124,8 +124,25 @@ def call_observer(
             updated_scale, updated_zero_point = observer(
                 value, g_idx=g_idx, global_scale=global_scale
             )
-            update_parameter_data(module, updated_scale, f"{base_name}_scale")
-            update_parameter_data(module, updated_zero_point, f"{base_name}_zero_point")
+            # register or update scale & zero_point parameters (supports block shapes)
+            scale_name = f"{base_name}_scale"
+            zp_name = f"{base_name}_zero_point"
+            if not hasattr(module, scale_name) or getattr(module, scale_name).shape != updated_scale.shape:
+                if hasattr(module, scale_name):
+                    delattr(module, scale_name)
+                module.register_parameter(
+                    scale_name, torch.nn.Parameter(updated_scale.clone())
+                )
```

Comment:
@kylesayrs do we need these (this and L141-143) to use `register_offload_parameter` instead?